### PR TITLE
Remove punit tests

### PR DESCRIPTION
**Pull request by maysamyabandeh**

### CODE REVIEW

File: build_tools/rocksdb-lego-determinator [hunk 1]
```diff
@@ -143,14 +143,14 @@ PARALLEL_UNIT_TEST_COMMANDS="[
...
             {
                 'name':'Build and test RocksDB debug version',
-                'shell':'$DEBUG make -j$(nproc) all && $SHM make check > /dev/null 2>&1 || $CONTRUN_NAME=punit_check $TASK_CREATION_TOOL || cat t/log-*',
+                'shell':'$DEBUG make -j$(nproc) all && ($SHM make check > /dev/null 2>&1 || $CONTRUN_NAME=punit_check $TASK_CREATION_TOOL) || cat t/log-*',
                 'user':'root',
                 $PARSER
             },
             $CLEANUP_ENV,
             {
                 'name':'Build and test RocksDB debug version under gcc-4.8.1',
```
Comment (yiwu-arbug) [inline build_tools/rocksdb-lego-determinator]: I think we dropped 4.8.1 support from us? How about remove the 4.8.1 tests?

**Comment by yiwu-arbug** [comment]
I assume compile error count as failure? But yeah I know it is a lot of noise..

File: build_tools/rocksdb-lego-determinator (hunk 1 above)
Comment (maysamyabandeh) [inline build_tools/rocksdb-lego-determinator]: @siying was mentioning it is important. I let him chime in.

**Comment by maysamyabandeh** [comment]
> I assume compile error count as failure? But yeah I know it is a lot of noise.

Yeah but in the test the compile did not actually fail. It should have been fine if the compile had returned the correct status but somehow it does not.

File: build_tools/rocksdb-lego-determinator (hunk 1 above)
Comment (siying) [inline build_tools/rocksdb-lego-determinator]: No we are not dropping 4.8.1 support. I can discuss offline.

Comment (siying) [inline build_tools/rocksdb-lego-determinator]: But I don't think we need to run punit in 4.8.1. We have UNIT_TEST_COMMANDS_481, so we should be good deleting this.

Comment (siying) [inline build_tools/rocksdb-lego-determinator]: https://www.facebook.com/groups/rocksdb.dev/permalink/1176824982416014/?comment_id=1211107005654478&comment_tracking=%7B%22tn%22%3A%22R%22%7D

Comment (maysamyabandeh) [inline build_tools/rocksdb-lego-determinator]: @siying you mean deleting the entire PARALLEL_UNIT_TEST_COMMANDS section or only the 2nd part that compiles with 4.8.1.?

Comment (siying) [inline build_tools/rocksdb-lego-determinator]: Yes we can delete PARALLEL_UNIT_TEST_COMMANDS . I don't think we need to test 4.8.1 so much.

**Pull request closed without merging**